main.tf

```diff
@@ -113,8 +113,9 @@ resource "aws_eip" "default" {
 
 # Apply the provisioner module for this resource
 module "ansible" {
-  source    = "git::https://github.com/cloudposse/tf_ansible.git?ref=tags/0.3.0"
+  source    = "git::https://github.com/cloudposse/tf_ansible.git?ref=tags/0.3.4"
   arguments = "${var.ansible_arguments}"
-  envs      = ["host=${aws_eip.default.public_ip}"]
+  envs      = "${concat(var.ansible_envs, list("host=${var.associate_public_ip_address ? aws_eip.default.public_ip : aws_instance.default.private_ip }"))}"
```

Comment:
I think we should use `public_dns` and `private_dns`. By using `public_dns` we can leverage split horizon DNS which is helpful when dealing with security group ingresses.